reactive/load_balancer.py

```diff
@@ -240,38 +242,46 @@ def provide_application_details():
             address = vips
         elif dns_record:
             address = dns_record
-    if address:
-        website.configure(port=hookenv.config('port'),
-                          private_address=address,
-                          hostname=address)
+        else:
+            address = hookenv.unit_get('public-address')
     else:
-        website.configure(port=hookenv.config('port'))
+        address = hookenv.unit_get('public-address')
+    return address
+
+
+@when('endpoint.lb-consumers.joined',
+      'leadership.is_leader')
+def provide_lb_consumers():
+    '''Respond to any LB requests via the lb-consumers relation.
+
+    This is used in favor for the more complex two relation setup using the
+    website and loadbalancer relations going forward.
+    '''
+    lb_consumers = endpoint_from_name('lb-consumers')
+    address = _get_lb_address()
+    for request in lb_consumers.all_requests:
+        request.response.address = address
+        lb_consumers.send_response(request)
+
+
+@when('website.available')
+def provide_application_details():
+    ''' re-use the nginx layer website relation to relay the hostname/port
+    to any consuming kubernetes-workers, or other units that require the
+    kubernetes API '''
+    website = endpoint_from_flag('website.available')
+    address = _get_lb_address()
+    website.configure(port=hookenv.config('port'),
+                      private_address=address,
+                      hostname=address)
```

Comment:
Without hacluster or loadbalancer-ips configured, this will tell other units to connect to a public-address, which is not ideal.

The correct address to use here is the ingress address of the website relation (per relation id). This was previously handled by omitting the private_address and hostname arguments, in which case, interface-http uses the proper ingress address: https://github.com/juju-solutions/interface-http/blob/632131b1f122daf6fb601fd4c9f1e4dbb1a92e09/provides.py#L31